<?xml version="1.0"?>
<launch>

    <!-- ****************** Arguments ******************** -->
    <arg name="robot_name"            default="$(env ROBOT_NAME)" />
    <arg name="base_frame"            default="base_footprint" />
    <arg name="output"                default="log" />
    <arg name="local_planner_type"    default="base"   doc="[base, dwa, teb]"/>
    <arg name="with_virtual_walls"    default="false" doc="Enables usage of virtual walls when set. Set to false when running SLAM." />


    <node pkg="move_base" type="move_base" respawn="true" name="move_base" output="$(arg output)" >
        <rosparam file="$(find leo_robot)/config/default_navigation/costmap_common_config.yaml" command="load" ns="global_costmap" subst_value="true" /> 
        <rosparam file="$(find leo_robot)/config/default_navigation/costmap_common_config.yaml" command="load" ns="local_costmap" subst_value="true" />
        <rosparam file="$(find leo_robot)/config/default_navigation/local_costmap_config.yaml"  command="load" />
        <rosparam file="$(find leo_robot)/config/default_navigation/global_costmap_config.yaml" command="load" />
        <rosparam file="$(find leo_robot)/config/default_navigation/$(arg local_planner_type)_local_planner_config.yaml" command="load"/>
        <rosparam file="$(find leo_robot)/config/default_navigation/move_base_config.yaml" command="load" /> 
        <rosparam file="$(find leo_robot)/config/default_navigation/global_planner_config.yaml" command="load"/>
        <rosparam file="$(find leo_robot)/config/default_navigation/costmap_global_params_plugins_virtual_walls.yaml" command="load" if="$(arg with_virtual_walls)" />
        <param name="base_global_planner" value="global_planner/GlobalPlanner" />
        <!-- <param name="base_global_planner" value="navfn/NavfnROS" /> -->
        <remap from="/$(arg robot_name)/odom" to="/$(arg robot_name)/odometry"/>
    </node> 
</launch>
